compiler/z/codegen/S390Instruction.cpp

```diff
@@ -1806,6 +1806,35 @@ TR::S390RILInstruction::adjustCallOffsetWithTrampoline(int32_t offset, uint8_t *
    return offsetHalfWords;
    }
 
+void
+TR::S390RILInstruction::addMetaDataForCodeAddress(uint8_t *cursor)
+   {
+   uintptr_t immediateAsAddress = (uintptr_t) (uint32_t) getSourceImmediate();
+   if (getNext()->getOpCode().getOpCodeValue() == TR::InstOpCode::IIHF)
+      immediateAsAddress |= ((uintptr_t) toS390RILInstruction(getNext())->getSourceImmediate()) << 32;
+
+   TR::Compilation *comp = cg()->comp();
+   bool staticPIC = std::find(comp->getStaticPICSites()->begin(), comp->getStaticPICSites()->end(), this) != comp->getStaticPICSites()->end();
+   bool staticMethodPIC = std::find(comp->getStaticMethodPICSites()->begin(), comp->getStaticMethodPICSites()->end(), this) != comp->getStaticMethodPICSites()->end();
+   bool staticHCRPIC = std::find(comp->getStaticHCRPICSites()->begin(), comp->getStaticHCRPICSites()->end(), this) != comp->getStaticHCRPICSites()->end();
```

Comment:
We do not need to search for both staticPICSites and StaticMethodPICSites. If we have it in one, we can not have it in another right ??